modular_skyrat/master_files/code/modules/clothing/shoes/boots.dm

```diff
@@ -1,3 +1,24 @@
 /obj/item/clothing/shoes/workboots/mining
 	icon = 'modular_skyrat/master_files/icons/obj/clothing/shoes.dmi' // To keep the old version before #8911
 	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/feet.dmi' // To keep the old version before #8911
+
+/obj/item/clothing/shoes/retro/shoes
+	name = "Retro Shoes"
+	desc = "A Post Postmodern take on a Retrofuturistic garment how innovative!"
+	icon = 'modular_skyrat/master_files/icons/obj/clothing/shoes.dmi'
+	icon_state = "retro_shoes"
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/feet.dmi'
+
+/obj/item/clothing/shoes/retro/boots
+	name = "Retro Boots"
+	desc = "A Post-Postmodern take on a Retrofuturistic garment, how innovative!"
+	icon = 'modular_skyrat/master_files/icons/obj/clothing/shoes.dmi'
+	icon_state = "retro_boots"
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/feet.dmi'
+
+/obj/item/clothing/shoes/retro/kneeboots
+	name = "Retro Kneeboots"
+	desc = "A Post-Postmodern take on a Retrofuturistic garment, how innovative!"
+	icon = 'modular_skyrat/master_files/icons/obj/clothing/shoes.dmi'
+	icon_state = "retro_kneeboots"
+	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/feet.dmi'
```

Comment:
With these all being subtypes of shoes/retro, I'd recommend doing this:

```suggestion
/obj/item/clothing/shoes/retro
	icon = 'modular_skyrat/master_files/icons/obj/clothing/shoes.dmi'
	worn_icon = 'modular_skyrat/master_files/icons/mob/clothing/feet.dmi'
	name = "Retro Shoes"
	desc = "A Post Postmodern take on a Retrofuturistic garment how innovative!"
	icon_state = "retro_shoes"

/obj/item/clothing/shoes/retro/boots
	name = "Retro Boots"
	icon_state = "retro_boots"

/obj/item/clothing/shoes/retro/kneeboots
	name = "Retro Kneeboots"
	icon_state = "retro_kneeboots"
```

Making the default (plain retro shoes in this case) the base type means we don't have to re-define any variables that would be duplicated anyways, specifically icon, worn_icon, and even desc - instead, all we need to do is re-define what we DO want to change, in these cases, icon_state and name. Functionally it's all the same as they're all shoes/retro, but shoes/retro/x will be defined to have their own unique appearance.
Does that make sense?